subcommand/server-acl-init/command_test.go

```diff
@@ -1312,6 +1488,156 @@ func TestRun_HTTPS(t *testing.T) {
 	require.True(ok)
 }
 
+// Test that the ACL replication token created from the primary DC can be used
+// for replication in the secondary DC.
+func TestRun_ACLReplicationTokenValid(t *testing.T) {
+	t.Parallel()
+
+	// Set up the primary DC first.
+	primarySvr, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.ACL.Enabled = true
+	})
+	require.NoError(t, err)
+	defer primarySvr.Stop()
+
+	primaryK8s := fake.NewSimpleClientset()
+	createTestK8SResources(t, primaryK8s, primarySvr.HTTPAddr, resourcePrefix, "http", ns)
+	require.NoError(t, err)
+
+	// Run the command to bootstrap ACLs
+	primaryUI := cli.NewMockUi()
+	primaryCmd := Command{
+		UI:        primaryUI,
+		clientset: primaryK8s,
+	}
+	primaryCmd.init()
+	primaryCmdArgs := []string{
+		"-k8s-namespace=" + ns,
+		"-expected-replicas=1",
+		"-server-label-selector=component=server,app=consul,release=" + releaseName,
+		"-resource-prefix=" + resourcePrefix,
+		"-create-acl-replication-token",
+	}
+	responseCode := primaryCmd.Run(primaryCmdArgs)
+	require.Equal(t, 0, responseCode, primaryUI.ErrorWriter.String())
+
+	// Retrieve the replication ACL token from the kubernetes secret.
+	tokenSecret, err := primaryK8s.CoreV1().Secrets(ns).Get("release-name-consul-acl-replication-acl-token", metav1.GetOptions{})
+	require.NoError(t, err)
+	require.NotNil(t, tokenSecret)
+	aclReplicationTokenBytes, ok := tokenSecret.Data["token"]
+	require.True(t, ok)
+	aclReplicationToken := string(aclReplicationTokenBytes)
+
+	// Now we can set up the secondary DC.
+	secondarySvr, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.Datacenter = "dc2"
+		c.ACL.Enabled = true
+		c.ACL.TokenReplication = true
+		c.PrimaryDatacenter = "dc1"
+		// When running via the Helm chart these are passed in to the servers.
+		c.ACL.Tokens.Agent = aclReplicationToken
+		c.ACL.Tokens.Replication = aclReplicationToken
+	})
+	require.NoError(t, err)
+	defer secondarySvr.Stop()
+
+	// When running via Helm, the WAN join config is set in the server config
+	// however that config isn't exposed by the testutil server so instead we need to
+	// make the join API call. This requires the bootstrap token permissions.
+	bootToken := getBootToken(t, primaryK8s, resourcePrefix, ns)
+	secondaryConsulClient, err := api.NewClient(&api.Config{
+		Address: secondarySvr.HTTPAddr,
+		Token:   bootToken,
```

Comment:
Does it make sense to try and test it with a token that has replication rules to make sure it works?

Reply:
So to be clear, this client is only being used to:
* make the join API call
* make calls to Consul to verify what was created
It isn't used by the consul server in the secondary dc nor the server-acl-init command.

So due to the above, the token we use here isn't really important. We can't use the replication token because it doesn't have the necessary permissions to make the join API call. I've commend on this above:
```
	// When running via Helm, the WAN join config is set in the server config
	// however that config isn't exposed by the testutil server so instead we need to
	// make the join API call. This requires the bootstrap token permissions.
```

But I'll move this comment to be on this line.

Reply:
Got it! Yes, on re-reading the code it makes more sense to me now.